src/Storage/Device/S3.php

```diff
@@ -67,7 +97,19 @@ public function upload($source, $path):bool
      */
     public function read(string $path):string
     {
-        return '';
+        $this->__resetResponse();
+        $verb = 'GET';
+        $uri = $path;
+        $uri = $uri !== '' ? '/'.str_replace('%2F', '/', rawurlencode($uri)) : '/';
```

Comment:
Might be more readable: `$uri !== '' ? '/'.str_replace('%2F', '/', rawurlencode($uri)) : '/';` - why not parse $path directly?

might be better to use: https://www.php.net/manual/en/function.urlencode.php?